src/Symfony/Component/Serializer/Normalizer/ScalarDenormalizer.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Encoder\XmlEncoder;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+final class ScalarDenormalizer implements DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    public const SCALAR_TYPES = [
+        Type::BUILTIN_TYPE_INT => true,
```

Comment:
Usage constants from `symfony/property-info` will require to be installed components always.
`symfony/serializer` uses property info as an optional dependency for now, isn't it?

Reply:
You're right, which is strange since both `symfony/property-access` and `symfony/property-info` are both used in other places like `Serializer` and `AbstractObjectNormalizer` classes...

Reply:
the property-info component is optional in AbstractObjectNormalizer

Reply:
I think I don't understand what you mean by optional then:
https://github.com/symfony/symfony/blob/49eafee03323253328d76ec9cb4c515f94d2f202/src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php#L17

Does not that mean that without the `PropertyInfo` component the class will not be found ?

Reply:
a use statement does not require the class to exist. Resolving a use statement happens purely at compile-time, without loading the class.
And in AbstractObjectNormalizer, usages of `Type` constants happen only in code paths where the type extractor is not null (at which point we can be sure that the component is installed)

Reply:
Good to know, so even the type hint for type extractor does not trigger the check for the class is we pass null I guess.
Thanks for the explanation.

Reply:
yeah, an `instanceof` check (and typehints are applying such checks) does not trigger any autoloading in PHP. They check the passed object.